core/src/main/java/com/datastax/oss/driver/internal/core/config/cloud/CloudConfigFactory.java

```diff
@@ -228,12 +230,22 @@ protected TrustManagerFactory createTrustManagerFactory(
   @NonNull
   protected BufferedReader fetchProxyMetadata(
       @NonNull URL metadataServiceUrl, @NonNull SSLContext sslContext) throws IOException {
-    HttpsURLConnection connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
-    connection.setSSLSocketFactory(sslContext.getSocketFactory());
-    connection.setRequestMethod("GET");
-    connection.setRequestProperty("host", "localhost");
-    return new BufferedReader(
-        new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+    try {
+      HttpsURLConnection connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
+      connection.setSSLSocketFactory(sslContext.getSocketFactory());
+      connection.setRequestMethod("GET");
+      connection.setRequestProperty("host", "localhost");
+      return new BufferedReader(
+          new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+    } catch (ConnectException e) {
+      throw new IllegalStateException(
```

Comment:
It's a matter of taste, but since this method already declares to throw `IOException` I think I would prefer to `throw new IOException("blah blah", e);`

Reply:
We can certainly change this however if you look over the rest of the CloudConfigFactory, typically when we run into something we throw an IllegalStateException. I was trying to keep some consistency with the rest o the behavior.